src/Context/Auth-context/Auth-context.js

```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
 const useAuth = () => useContext(authContext);
 
 const AuthProvider = ({ children }) => {
-  const [userData, setUserData] = useState([]);
+  const [credentialData, setCredentailData] = useState({
+    tokenData: localStorage.getItem("token"),
+    isAuth: localStorage.getItem("token") ? true : false,
+  });
+  const { tokenData, isAuth } = credentialData;
 
-  const loginHandler = async ({ email, password }) => {
+  const signupHandler = async (data) => {
     try {
-      const response = await axios.post(`/api/auth/login`, {
-        email: email,
-        password: password,
+      const response = await axios.post("/api/auth/signup", {
+        email: data.email,
+        password: data.password,
+        firstName: data.firstName,
+        lastName: data.lastName,
       });
```

Comment:
you can destructure these values from data itslef, ``` const response = await axios.post("/api/auth/signup", {
        email,
        password,
        firstName,
        lastName,
      }); ```
and then there will be no need of writing email:email .